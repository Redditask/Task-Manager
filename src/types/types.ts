import {FC} from "react";

export interface Task {
    id?: string;
    taskText: string;

    year?: number;
    month?: number;
    day?: number;
    startTime: Time;
    endTime: Time;
    color: string;
}

export interface Time {
    hour: number;
    min: number;
}

export interface Color {
    name: string;
    color: string;
}

export interface Route {
    path: string,
    Component: FC<{}>;
}

export interface CustomDate {
    year: number;
    month: number;
    day: number;
}

export interface User {
    exp: number;
    iat: number;
    id: number;
    login: string;
}

export type StringChangeEvent = {
    target: {
        value: string
    }
};

export type Theme = "light" | "dark";
