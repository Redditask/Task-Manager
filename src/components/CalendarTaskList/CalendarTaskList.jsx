import styles from "./CalendarTaskList.module.scss";

import React from 'react';
import {useSelector} from "react-redux";
import {selectTasks} from "../../store/selectors";
import CalendarTask from "../CalendarTask/CalendarTask";

const CalendarTaskList = ({data, setDropTask}) => {
    const tasks = useSelector(selectTasks);

    return (
        <div className={styles.CalendarTaskList}>
            {
                tasks.map((task, index) => {
                    if (task.day === data.day
                        && task.month === data.month
                        && task.year === data.year)
                        return (
                            <CalendarTask
                                key={task.taskText + index}
                                task={task}
                                setDropTask={setDropTask}
                            />
                        )
                })
            }
        </div>
    );
};

export default CalendarTaskList;
