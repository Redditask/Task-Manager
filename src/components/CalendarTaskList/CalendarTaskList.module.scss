.CalendarTaskList{
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;

  overflow: auto;
  &::-webkit-scrollbar{
    border: none;

    &-thumb{
      border: none;
    }
  }

  gap: 1rem;
}
