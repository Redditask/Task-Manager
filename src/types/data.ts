export type Task = {
    id?: string;
    taskText: string;
    year?: number;
    month?: number;
    day?: number;
    startTime: Time;
    endTime: Time;
    color: string;
};

export type Time = {
    hour: number;
    min: number;
};

export type Color = {
    name: string;
    color: string;
};

export type CustomDate = {
    year: number;
    month: number;
    day: number;
};

export type StringChangeEvent = {
    target: {
        value: string
    }
};

export type Theme = "light" | "dark";
