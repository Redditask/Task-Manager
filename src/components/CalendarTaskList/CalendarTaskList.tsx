import styles from "./CalendarTaskList.module.scss";

import React from 'react';

import {useAppSelector} from "../../hooks/hooks";
import {selectTasks} from "../../store/selectors";

import CalendarTask from "../CalendarTask/CalendarTask";

import {CustomDate, Task} from "../../types/types";

interface CalendarTaskListProps {
    date: CustomDate;
    setDropTask: (dropTask: Task) => void;
}

const CalendarTaskList:React.FC<CalendarTaskListProps> = ({date, setDropTask}) => {
    const tasks: Task[] = useAppSelector(selectTasks);

    return (
        <div className={styles.CalendarTaskList}>
            {
                tasks.map((task, index) => {
                    if (task.day === date.day
                        && task.month === date.month
                        && task.year === date.year)
                        return (
                            <CalendarTask
                                key={task.taskText + index}
                                task={task}
                                setDropTask={setDropTask}
                            />
                        )
                })
            }
        </div>
    );
};

export default CalendarTaskList;
