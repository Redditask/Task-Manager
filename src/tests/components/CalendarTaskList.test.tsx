import {render} from "@testing-library/react";
import * as ReduxHooks from "react-redux";

import CalendarTaskList from "../../components/CalendarTaskList/CalendarTaskList";

import {Task} from "../../types/data";

jest.mock("react-redux");

const mockedUseSelector = jest.spyOn(ReduxHooks, "useSelector");

const someTasks: Task[] = [
    {
        id: "2021-11-27T16:33:22.812Z",
        taskText: "Your task",
        year: 2022,
        month: 5,
        day: 17,
        startTime: {hour: 3, min: 14},
        endTime: {hour: 23, min: 5},
        color: "beige"
    },
    {
        id: "2022-5-17T16:33:22.812Z",
        taskText: "Your task",
        year: 2022,
        month: 5,
        day: 17,
        startTime: {hour: 13, min: 8},
        endTime: {hour: 14, min: 25},
        color: "#00FF7F"
    }
];

describe("CalendarTaskList", ()=>{
    it("should create TaskList with empty task list", ()=>{
        mockedUseSelector.mockReturnValue([]);

        const component = render(<CalendarTaskList
            data={{day: 0, month: 0, year: 0}}
            setDropTask={jest.fn()}
        />);

        expect(component).toMatchSnapshot();
    });

    it("should create TaskList with tasks", ()=>{
        mockedUseSelector.mockReturnValue(someTasks);
        
        const component = render(
            <CalendarTaskList
                data={{day: 17, month: 5, year: 2022}}
                setDropTask={jest.fn()}
            />
        );

        expect(component).toMatchSnapshot();
    });
});
