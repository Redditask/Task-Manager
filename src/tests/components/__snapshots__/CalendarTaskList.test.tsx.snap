// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarTaskList should create TaskList with empty task list 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="CalendarTaskList"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="CalendarTaskList"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CalendarTaskList should create TaskList with tasks 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="CalendarTaskList"
      >
        <div
          class="CalendarTask"
          draggable="true"
          style="background-color: rgb(0, 255, 127);"
        >
          Your task 2
        </div>
        <div
          class="CalendarTask"
          draggable="true"
          style="background-color: beige;"
        >
          Your task 3
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="CalendarTaskList"
    >
      <div
        class="CalendarTask"
        draggable="true"
        style="background-color: rgb(0, 255, 127);"
      >
        Your task 2
      </div>
      <div
        class="CalendarTask"
        draggable="true"
        style="background-color: beige;"
      >
        Your task 3
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
